src/data/moves/move.ts

```diff
@@ -619,12 +619,30 @@ export default class Move implements Localizable {
 
   /**
    * Checks if the move flag applies to the pokemon(s) using/receiving the move
+   *
+   * This method will take the `user`'s ability into account when reporting flags, e.g.
+   * calling this method for {@linkcode MoveFlags.MAKES_CONTACT | MAKES_CONTACT}
+   * will return `false` if the user has a {@linkcode Abilities.LONG_REACH} that is not being suppressed.
+   *
+   * **Note:** This method only checks if the move should have effectively have the flag applied to its use.
+   * It does *not* check whether the flag will trigger related effects.
+   * For example using this method to check {@linkcode MoveFlags.WIND_MOVE}
+   * will not consider {@linkcode Abilities.WIND_RIDER | Wind Rider }.
+   *
+   * To simply check whether the move has a flag, use {@linkcode hasFlag}.
    * @param flag {@linkcode MoveFlags} MoveFlag to check on user and/or target
    * @param user {@linkcode Pokemon} the Pokemon using the move
    * @param target {@linkcode Pokemon} the Pokemon receiving the move
+   * @param isFollowUp (defaults to `false) `true` if the move was used as a follow up
    * @returns boolean
-   */
-  checkFlag(flag: MoveFlags, user: Pokemon, target: Pokemon | null): boolean {
+   * @see {@linkcode hasFlag}
+   */
+  doesFlagEffectApply({ flag, user, target = null, isFollowUp = false }: {
+    flag: MoveFlags;
+    user: Pokemon;
+    target?: Pokemon | null;
```

Comment:
```suggestion
    target?: Pokemon;
```
Also change `Move#hitsSubstitute()`'s `target` param to be optional instead of `| null` (that's the reason this param is `| null`, but it's unnecessary).